### ISSUE TITLE
Skybox - Mirror and Rotate parameters not being considered unique to each CubeMap node.

### ISSUE THREAD

**Issue by Poodmund** [Sigma88/Sigma-Replacements#25]
I have 2 skyboxes installed, GPP's and one of my own, both just adding their SkyBox node in the normal way.

```
@SigmaReplacements
{
    SkyBox
    {
        CubeMap
        {
			mirror = true
			rotate = true
            SkyBox = GPP/GPP_Skybox/Galileo
        }
    }
}
```
and
```
@SigmaReplacements:FINAL
{
    SkyBox
    {
        CubeMap
        {
			mirror = false
			rotate = false
            SkyBox = PoodsSkyboxes/DeepStarMap
        }
    }
}
```

GPP's config has 'mirror = true' and 'rotate = true' but my config has 'mirror = false' and 'rotate = false' yet with both skyboxes loaded it seems that GPP's true statements take precedence and whenever my skybox is chosen to be displayed for a game save, it is displayed either mirrored, rotated or both.

---
In the logs attached I did the following in-game:
- Loaded KSP with only SR: Skybox v0.1.4, Module Manager 3.0.1 and the two skyboxes
- Created a new Career Save called Skybox. Went to the Tracking Station where the GPP Skybox was displayed.
- Quit to the Main Menu
- Created a new Sandbox Save called Skybox 1. Went the to Tracking Station where my skybox was displayed... but it displayed rotated and/or mirrored.
- Quit to Main Menu and closed the game.
---
Logs: https://www.dropbox.com/s/6k13luxnwtsn5jl/Logs.zip?dl=0

**Comment by Sigma88** [comment]
yes that's the "intended behaviour" but it's dumb so I'll rewrite it

thanks for remembering me of this :)

**Comment by Sigma88** [comment]
I forced the generation of 160000 game seeds and while there are some stretches of the same choice being made, overall the percentages match a random process.

these are the counts I get on 160000 seeds:

Option Chosen | Out of 2 | Out of 3 | Out of 4 | Out of 5 | Out of 6 | Out of 7
-- | -- | -- | -- | -- | -- | --
0 | 79283 | 53435 | 39416 | 31945 | 26509 | 22836
1 | 80717 | 52986 | 39976 | 32105 | 26677 | 22915
2 | 0 | 53579 | 39867 | 32024 | 26465 | 22882
3 | 0 | 0 | 40741 | 31936 | 26926 | 22607
4 | 0 | 0 | 0 | 31990 | 26309 | 22918
5 | 0 | 0 | 0 | 0 | 27114 | 23063
6 | 0 | 0 | 0 | 0 | 0 | 22779
7 | 0 | 0 | 0 | 0 | 0 | 0


these are the %

Option Chosen | Out of 2 | Out of 3 | Out of 4 | Out of 5 | Out of 6 | Out of 7
-- | -- | -- | -- | -- | -- | --
0 | 50% | 33% | 25% | 20% | 17% | 14%
1 | 50% | 33% | 25% | 20% | 17% | 14%
2 | 0% | 33% | 25% | 20% | 17% | 14%
3 | 0% | 0% | 25% | 20% | 17% | 14%
4 | 0% | 0% | 0% | 20% | 16% | 14%
5 | 0% | 0% | 0% | 0% | 17% | 14%
6 | 0% | 0% | 0% | 0% | 0% | 14%
7 | 0% | 0% | 0% | 0% | 0% | 0%



so basically you were a bit unlucky and got a stretch of mostly one result

**Comment by Sigma88** [comment]
just to give you an idea, if I limit my analysis to the first 30 seeds these are the results


Option Chosen | Out of 2 | Out of 3 | Out of 4 | Out of 5 | Out of 6 | Out of 7
-- | -- | -- | -- | -- | -- | --
0 | 10 | 7 | 5 | 4 | 1 | 5
1 | 20 | 7 | 9 | 4 | 5 | 3
2 | 0 | 16 | 5 | 11 | 7 | 2
3 | 0 | 0 | 11 | 7 | 6 | 6
4 | 0 | 0 | 0 | 4 | 2 | 5
5 | 0 | 0 | 0 | 0 | 9 | 3
6 | 0 | 0 | 0 | 0 | 0 | 6
7 | 0 | 0 | 0 | 0 | 0 | 0

and with %


Option Chosen | Out of 2 | Out of 3 | Out of 4 | Out of 5 | Out of 6 | Out of 7
-- | -- | -- | -- | -- | -- | --
0 | 33% | 23% | 17% | 13% | 3% | 17%
1 | 67% | 23% | 30% | 13% | 17% | 10%
2 | 0% | 53% | 17% | 37% | 23% | 7%
3 | 0% | 0% | 37% | 23% | 20% | 20%
4 | 0% | 0% | 0% | 13% | 7% | 17%
5 | 0% | 0% | 0% | 0% | 30% | 10%
6 | 0% | 0% | 0% | 0% | 0% | 20%
7 | 0% | 0% | 0% | 0% | 0% | 0%


I don't think there are solutions to this to be honest

**Comment by Poodmund** [comment]
Random is random and as such if you expect pure equality then you might as well not use an RNG like process for determination. Your results look as though everything is working as it should be from a choice of Skybox point of view, however, the bigger issue here is that the Mirror and Rotate rules are not being respected and that was the main point of the issue I raised.

**Comment by Sigma88** [comment]
Yeah sure, that behaviour was just badly implemented.

And regarding random behaviour. I don't really want random. I'd rather have an homogeneous number of events for each possible choice. But I'm too lazy to implement that :)

**Comment by Sigma88** [comment]
fixed and released
